bin/test-workflow/7_app_deploy.sh

```diff
@@ -0,0 +1,285 @@
+  announce "Deploying test apps for $TEST_APP_NAMESPACE_NAME."
+
+  URLENCODED_AUTHN_ID=$(urlencode $AUTHENTICATOR_ID)
+
+  set_namespace $TEST_APP_NAMESPACE_NAME
+  init_registry_creds
+  init_connection_specs
+
+  if is_minienv; then
+    IMAGE_PULL_POLICY='Never'
+  else
+    IMAGE_PULL_POLICY='Always'
+  fi
+
+  deploy_app_backend
+  # deploy_secretless_app
+  deploy_sidecar_app
+  # deploy_init_container_app
+  # deploy_init_container_app_with_host_outside_apps
+}
+
+###########################
+init_registry_creds() {
+  if [[ "${PLATFORM}" == "kubernetes" ]] && [[ -n "${DOCKER_EMAIL}" ]]; then
+    announce "Creating image pull secret."
+
+    kubectl delete --ignore-not-found secret dockerpullsecret
+
+    kubectl create secret docker-registry dockerpullsecret \
+      --docker-server=${PULL_DOCKER_REGISTRY_URL} \
+      --docker-username=$DOCKER_USERNAME \
+      --docker-password=$DOCKER_PASSWORD \
+      --docker-email=$DOCKER_EMAIL
+  elif [[ "$PLATFORM" == "openshift" ]]; then
+    announce "Creating image pull secret."
+
+    $cli delete --ignore-not-found secrets dockerpullsecret
+
+    $cli secrets new-dockercfg dockerpullsecret \
+      --docker-server=${PULL_DOCKER_REGISTRY_URL} \
+      --docker-username=_ \
+      --docker-password=$($cli whoami -t) \
+      --docker-email=_
+
+    $cli secrets add serviceaccount/default secrets/dockerpullsecret --for=pull
+  fi
+}
+
+###########################
+init_connection_specs() {
+  test_sidecar_app_docker_image=$(platform_image_for_pull test-sidecar-app)
+  test_init_app_docker_image=$(platform_image_for_pull test-init-app)
+
+  if [[ "$LOCAL_AUTHENTICATOR" == "true" ]]; then
+    authenticator_client_image=$(platform_image_for_pull conjur-authn-k8s-client)
+    secretless_image=$(platform_image_for_pull secretless-broker)
+  else
+    authenticator_client_image="cyberark/conjur-authn-k8s-client"
+    secretless_image="cyberark/secretless-broker"
+  fi
+
+  if [[ "$CONJUR_OSS_HELM_INSTALLED" == "true" ]]; then
+    conjur_appliance_url=${CONJUR_APPLIANCE_URL:-https://conjur-oss.$CONJUR_NAMESPACE.svc.cluster.local}
+  else
+    conjur_follower_name=${CONJUR_FOLLOWER_NAME:-conjur-follower}
+    conjur_appliance_url=https://$conjur_follower_name.$CONJUR_NAMESPACE.svc.cluster.local/api
+  fi
+  conjur_authenticator_url="$conjur_appliance_url/authn-k8s/$URLENCODED_AUTHN_ID"
+}
+
+###########################
+deploy_app_backend() {
+  $cli delete --ignore-not-found \
+     service/test-summon-init-app-backend \
+     service/test-summon-sidecar-app-backend \
+     service/test-secretless-app-backend \
+     statefulset/summon-init-pg \
+     statefulset/secretless-pg \
+     statefulset/summon-init-mysql \
+     statefulset/summon-sidecar-mysql \
+     statefulset/secretless-mysql \
+     secret/test-app-backend-certs
+
+  ensure_env_database
+  case "${TEST_APP_DATABASE}" in
+  postgres)
+    echo "Create secrets for test app backend"
+    $cli --namespace $TEST_APP_NAMESPACE_NAME \
```

Comment:
I think the creation of a Kubernetes Secret to store the CA cert & key can be moved to the parent application deploy Helm chart (the `ca.pem` and `ca-key.pem` files would get moved e.g. to a `files/` subdirectory in that chart.

The advantage of doing this in the parent Helm chart would be that we would have to handle this in the scripts, and the Secret would get deleted via `helm delete ...`.

Reply:
Or another option would be to add a Kubernetes Secret for test app backends in each of the Helm subcharts, but they'd each need unique names.

Reply:
Added the secret creation to `app-namespace-prep` chart since it needs to be present before installing sample app backend

Reply:
Reverted creation of Secret in a helm chart after discussion with @diverdane